llvm/test/CodeGen/RISCV/addimm-mulimm.ll

```diff
@@ -561,8 +562,8 @@ define i64 @add_mul_combine_infinite_loop(i64 %x) {
 ; RV64IMB:       # %bb.0:
 ; RV64IMB-NEXT:    addi a0, a0, 86
 ; RV64IMB-NEXT:    sh1add a0, a0, a0
-; RV64IMB-NEXT:    li a1, -16
-; RV64IMB-NEXT:    sh3add a0, a0, a1
+; RV64IMB-NEXT:    slli a0, a0, 3
```

Comment:
This seems like a mild regression, but I think only indirectly related to this patch.

Reply:
Why do we end up with this instead of `(x*3+256)*8`?

Reply:
This looks to be the result of transformAddImmMulImm.  I agree that looks like a questionable transform.